bodyparsers/msgpack/README.md

```diff
@@ -0,0 +1,185 @@
+import {post, getModelSchemaRef, requestBody} from '@loopback/rest';
+
+class TodoController {
+  // Omitted constructor for bevity
+
+  @post('/todos')
+  async create(
+    @requestBody({
+      content: {
+        // Change existing or append a new requset body accepted MIME type
+        'application/msgpack': {
+          schema: getModelSchemaRef(Todo, {
+            title: 'NewTodo',
+            exclude: ['id'],
+          }),
+        },
+      },
+    })
+    todo: // Keep the response body object type, since the body parser transparently
+    // converts it into a JavaScript object.
+    Omit<Todo, 'id'>,
+
+    // For bevity, the function does not return anything. See
+    // 'Returning MessagePack Requests' below.
+  ): void {
+    this.todoRepository.create(todo);
+  }
+}
+```
+
+The MessagePack request payload will be transparently converted into a
+JavaScript object and validated against the JSON Schema.
+
+### Returning MessagePack Requests
+
+{% include note.html content="The body parser will not convert responses into `application/msgpack`. This needs to be done manually." %}
+
+To return MessagePack requests in a controller, amend the requestBody decorator
+to include the MIME type as a possible response and user a parser library.
+
+For example, to update the Todo controller to return in MessagePack:
+
+```ts
+import {
+  MsgPackBodyParserBindings,
+  MsgPackParserLibrary,
+} from '@loopback/bodyparser-msgpack';
+import {inject} from '@loopback/core';
+import {getModelSchemaRef, post, Response, RestBindings} from '@loopback/rest';
+
+// It's recommended to bind an instance to context, then inject it into the
+// controller. It's imported here for bevity.
+import msgpack from 'msgpack';
+
+class TodoController {
+  private readonly _response: Response;
+
+  constructor(
+    // Omitted other dependency injections (e.g. repository) for bevity.
+
+    // Inject the Response object to the controller
+    @inject(RestBindings.Http.RESPONSE)
+    private readonly _res: Response,
+  ) {}
+
+  @get('/todos', {
+    responses: {
+      '200': {
+        description: 'Array of Todo model instances',
+        content: {
+          // Update existing or amend new possible response
+          'application/msgpath': {
+            schema: {type: 'array', items: getModelSchemaRef(Todo)},
+          },
+        },
+      },
+    },
+  })
+  async findTodos(
+    @param.filter(Todo)
+    filter?: Filter<Todo>,
+
+    // Change function return type to Promise<void>.
+  ): Promise<void> {
+    // Internally, LoopBack 4 will try to guess and override the `Content-Type`
+    // header, even after manually setting the headers.
+    // Buffers are automatically detected as `application/octet-stream`.
+    // We can use `Response.end()` to bypass that.
+    //
+    // See: https://github.com/strongloop/loopback-next/issues/436
+    //
+    this._res
+      .type('application/msgpack')
+      .end(msgpack.encode(this.todoRepository.find(filter)));
```

Comment:
Can you please update `import` statements in this example to show where is `msgpack` imported from? 

I think it would be best if our users can use the same version of msgpack that's used for parsing the requests, i.e.

```ts
import {msgpack} from '@loopback/bodyparser-msgpack';
```

Reply:
Also in the light of the discussion in https://github.com/strongloop/loopback-next/pull/6155/files#r481194436, maybe we should rename the extension to `@loopback/rest-msgpack`? (In the future, we may want to add msgpack support for other transports like messaging, so the `rest-` prefix should make it clear this extension is for REST.)

Reply:
> maybe we should rename the extension to `@loopback/rest-msgpack`

Just an idea to consider, maybe it's too early considering that this package will provide just the body parser for quite some time until we implement support for content-type negotiation in the framework.

Reply:
I think it makes sense to rename it to `@loopback/rest-msgpack`.